DSCResources/MSFT_xDhcpServerExclusionRange/MSFT_xDhcpServerExclusionRange.psm1

```diff
@@ -0,0 +1,319 @@
+Import-Module $PSScriptRoot\..\Helper.psm1 -Verbose:$false
+
+# Localized messages
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+AddingExclusionRangeMessage   = Adding DHCP server exclusion range with the given ScopeId...
+CheckExclusionRangeMessage    = Checking DHCP server exclusion range with the given ScopeId...
+SetExclusionRangeMessage      = DHCP server exclusion range with ScopeId {0} is now present
+RemovingExclusionRangeMessage = Removing DHCP server exclusion range with the given ScopeId...
+DeleteExclusionRangeMessage   = DHCP server exclusion range with the given ScopeId is now absent
+TestExclusionRangeMessage     = DHCP server exclusion range with the given ScopeId is {0} and it should be {1}
+
+CheckPropertyMessage      = Checking DHCP server scope {0} ...
+NotDesiredPropertyMessage = DHCP server scope {0} is not correct. Expected {1}, actual {2}
+DesiredPropertyMessage    = DHCP server scope {0} is correct.
+SetPropertyMessage        = DHCP server scope {0} is set to {1}.
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory)]
+        [String]$ScopeId,
+
+        [parameter(Mandatory)]
+        [String]$StartRange,
+
+        [parameter(Mandatory)]
+        [String]$EndRange,
+
+        [ValidateSet('IPv4')]
+        [String]$AddressFamily = 'IPv4'
+
+    )
+#region Input Validation
+
+    # Check for DhcpServer module/role
+    Assert-Module -moduleName DHCPServer
+
+    # Convert the ScopeId to be a valid IPAddress
+    $ScopeId = (Get-ValidIpAddress -ipString $ScopeId -AddressFamily $AddressFamily -parameterName 'ScopeId').ToString()
+
+    # Convert the Start Range to be a valid IPAddress
+    $StartRange = (Get-ValidIpAddress -ipString $StartRange -AddressFamily $AddressFamily -parameterName 'StartRange').ToString()
+
+    # Convert the End Range to be a valid IPAddress
+    $EndRange = (Get-ValidIpAddress -ipString $EndRange -AddressFamily $AddressFamily -parameterName 'EndRange').ToString()
+
+    # Check to ensure startRange is smaller than endRange
+    if($EndRange.Address -lt $StartRange.Address)
+    {
+        $errorMsg = $LocalizedData.InvalidStartAndEndRangeMessage
+        New-TerminatingError -errorId RangeNotCorrect -errorMessage $errorMsg -errorCategory InvalidArgument
+    }
+
+#endregion Input Validation
+
+    if ($AddressFamily -eq "IPv4")
+    {
+      $dhcpExclusionRange = Get-DhcpServerv4ExclusionRange | ? {($_.ScopeId -eq $ScopeId) -and ($_.StartRange -eq $StartRange) -and ($_.EndRange -eq $EndRange)}
```

Comment:
Can you replace the alias with Where-Object and place on a new line to try an keep the line length below 100 chars as per the [contribution guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?